requirements/earliest.txt

```diff
@@ -1,5 +1,5 @@
 -r base.txt
-fsspec==2023.9.0
-azure-core==1.23.1
-azure-datalake-store==0.0.46
-azure-storage-blob==12.12.0
+fsspec==2023.12.0
+azure-core==1.28.0
+azure-datalake-store==0.0.53
+azure-storage-blob==12.17.0
```

Comment:
- [fsspec 2023.12.0](https://pypi.org/project/fsspec/2023.12.0/): officially supports up to `py311`
- [azure-storage-blob 12.17.0](https://pypi.org/project/azure-storage-blob/12.17.0/): officially supports up to `py310`

Reply:
Usually, they will support more recent too, but at the time of release they were not explicitly tested with a later python (they come quite frequently these days).